oci/extensions.bzl

```diff
@@ -1,6 +1,21 @@
 "extensions for bzlmod"
 
 load(":repositories.bzl", "oci_register_toolchains")
+load(":pull.bzl", "oci_pull")
+
+# TODO: it sucks that the API of the oci_pull macro has to be repeated here.
+pull = tag_class(attrs = {
+    "name": attr.string(doc = "Name of the generated repository"),
+    "image": attr.string(doc = """the remote image without a tag, such as gcr.io/bazel-public/bazel"""),
+    "platforms": attr.string_list(doc = """for multi-architecture images, a dictionary of the platforms it supports
+            This creates a separate external repository for each platform, avoiding fetching layers."""),
+    "digest": attr.string(doc = """the digest string, starting with "sha256:", "sha512:", etc.
+            If omitted, instructions for pinning are provided."""),
+    "tag": attr.string(doc = """a tag to choose an image from the registry.
+            Exactly one of `tag` and `digest` must be set.
+            Since tags are mutable, this is not reproducible, so a warning is printed."""),
+    "reproducible": attr.bool(doc = """Set to False to silence the warning about reproducibility when using `tag`.""", default = True),
+})
```

Comment:
Can you just share the `attrs` object with the rule? Then add any extra attributes for the macro (if there is one).

Reply:
https://github.com/aspect-build/rules_js/blob/main/npm/extensions.bzl#L157

Reply:
I tried that at first, however this calls a wrapper macro where the argument are nearly non-overlapping. For instance the underlying rule has a `platform` attribute, but the macro has a loop so it accepts `platforms` instead.